File: konversi nilai.py
def has_numbers(inputString):
     return any(char.isdigit() for char in inputString)
nama = str(input("Nama mahasiswa:  "))
nim = float(input("Nim: "))
p = float(input("Masukkan nilai partisipasi: "))
t = float(input("Masukkan nilai tugas: "))
uts = float(input("Masukkan nilai uts: "))
uas = float(input("Masukkan nilai uas: "))
score = ((2*p)+(3*t)+(2*uts)+(3*uas))/10
if nim <0 or p <0 or t<0 or uts <0 or uas <0: 
    raise ValueError
print("Your grade is : ",score)
if nim <0 or p <0 or t<0 or uts <0 or uas <0: 
    raise "Data tidak valid"
elif all(x.isalpha() or x.isspace() for x in nama):
    print("Your grade is : ",score)
elif has_numbers(nama);
    raise "Nama tidak valid"
else: raise ValueError

if score >= 90:
    Grade="A"
elif score >= 80:
    Grade="B"
elif score >= 70:
    Grade="C"
elif score >= 60:
    Grade="D"
else:
    Grade="E"

print("Your grade is : ",Grade)

File: konversi nilai.php
<?php
if(isset($_GET["Hitung"])){
	if ($nilai >=90){
	$hasil .="<br>A";
}
else if ($nilai >=80){
	$hasil .="<br>B";
}
else if ($nilai >=70){
	$hasil .="<br>C";
}
else if ($nilai >=90){
	$hasil .="<br>D";
}
else if ($nilai >=90){
	$hasil .="<br>E";
}
$p=$_GET['p'];
$t=$_GET['t'];
$uts=$_GET['uts'];
$uas=$_GET['uas'];
$hasil;
$nilai=((2*$p)+(3*$t)+(2*$uts)+(3*$uas))/10;
if ($p <0||$t <0||$uts <0||$uas <0){
			$hasil= "Data tidak terbaca<br>";
		} 
		else {
			$hasil= strval($nilai);

		}
}

?>

<!DOCTYPE html>
<html>
<head>
	<title>Nilai Mahasiswa Unesa</title>
	<style>
		body{
			background-image: url(bgg.jpeg);
            background-size: cover;
            background-repeat: no-repeat;
            background-position: center;
            background-attachment: fixed;
            height: 100%;
		}
	</style>
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.0/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-KyZXEAg3QhqLMpG8r+8fhAXLRk2vvoC2f3B09zVXn8CA5QIVfZOJ3BCsw2P0p/We" crossorigin="anonymous">
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.0/dist/js/bootstrap.bundle.min.js" integrity="sha384-U1DAWAznBHeqEIlVSCgzq+c9gqGAJn5c/t99JyeKa9xxaYpSvHU5awsuZVVFIhvj" crossorigin="anonymous"></script>
    <link rel="preconnect" href="https://fonts.googleapis.com">
	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
	<link href="https://fonts.googleapis.com/css2?family=Poly&display=swap" rel="stylesheet">
</head>
<body>
	<div align="center">
	<form action="" method="GET" >
	<h1>NILAI MAHASISWA UNESA</h1><br>
	<p>Partisipasi</p>
	<input type="partisipasi" name="p">
	<p>Tugas</p>
	<input type="tugas" name="t">
	<p>uts</p>
	<input type="uts" name="uts"><br>
	<p>uas</p>
	<input type="uas" name="uas"><br>
	<input type="submit" name="Hitung"  class="btn btn-primary">
    <button type="reset" class="btn btn-secondary">Reset</button>
	<br>
	<br>
	<br>
	<h1 align="center">
		<?php
		if(isset($_GET["Hitung"])){
		echo $hasil;
		}
		?>
		
	</h1>
</form>
</div>
</body>
</html>

